data/transactions/payment.go

```diff
@@ -42,8 +42,11 @@ func (payment PaymentTxnFields) checkSpender(header Header, spec SpecialAddresse
 		return fmt.Errorf("transaction cannot close account to its sender %v", header.Sender)
 	}
 
-	// the FeeSink account may only spend to the IncentivePool
+	// the FeeSink account may only spend to the IncentivePool (not at all, if EnableMining)
 	if header.Sender == spec.FeeSink {
+		if proto.EnableMining {
+			return fmt.Errorf("cannot spend from fee sink address %v", header.Sender)
```

Comment:
If the goal is to prevent fee "laundering", as I was getting at in https://github.com/algorand/go-algorand/pull/5757#discussion_r1471610064, then this is sufficient.

I'll leave this open just to learn what the intention was here, but I don't necessarily see a problem anymore.

Reply:
I wanted to make it always true that the `FeeSink` always had enough to pay "mining fees" (the payment that comes just from fees). So I had to a) not increment `FeesCollected` for fees "paid" by `FeeSink` (since the amount in the `FeeSink` does not grow from such a fee) and b) preclude payments from `FeeSink`.

In some ways, this is all less important now, because when I added bonus payments, I made sure the incentives never took the `FeeSink` below `MinBalance`. So it's sort of all moot.

I suppose I'm happy with the current status - `FeeSink` can't pay, and `FeeSink`, fees are not counted in `FeesCollected`, but other txns are possible.  But I don't think that's "right" in a deep sense. I'm open to other preferences.

Reply:
Thanks for the additional context. I'm comfortable with this